hugr-core/src/hugr/validate.rs

```diff
@@ -28,33 +27,28 @@ use super::ExtensionError;
 ///
 /// TODO: Consider implementing updatable dominator trees and storing it in the
 /// Hugr to avoid recomputing it every time.
-struct ValidationContext<'a, 'b> {
+struct ValidationContext<'a> {
     hugr: &'a Hugr,
     /// Dominator tree for each CFG region, using the container node as index.
     dominators: HashMap<Node, Dominators<Node>>,
-    /// Registry of available Extensions
-    extension_registry: &'b ExtensionRegistry,
 }
 
 impl Hugr {
     /// Check the validity of the HUGR, assuming that it has no open extension
     /// variables.
     /// TODO: Add a version of validation which allows for open extension
     /// variables (see github issue #457)
-    pub fn validate(&self, extension_registry: &ExtensionRegistry) -> Result<(), ValidationError> {
-        self.validate_no_extensions(extension_registry)?;
+    pub fn validate(&self) -> Result<(), ValidationError> {
+        self.validate_no_extensions()?;
         #[cfg(feature = "extension_inference")]
```

Comment:
It's nicer do `if cfg!(...)` since the call gets compiled (and dead-code eliminated), so any syntax error gets reported even when the feature is disabled.
I'll change it here.